docs/src/languages/nodejs/debug.md

```diff
@@ -3,121 +3,244 @@ title: "Debugging"
 weight: 1
 ---
 
-Effectively debugging web applications isn’t trivial, especially when an HTTP request goes through multiple layers until it reaches the web app, which is usually the case in cloud platforms.
+Effectively debugging web apps takes effort,
+especially when an HTTP request goes through multiple layers before reaching your web app.
+Follow the steps below to debug a specific app.
 
-{{< note >}}
-Whether you need to debug locally or remotely, always do so in a non-production environment.
-{{< /note >}}
+You can choose to debug in an environment deployed to Platform.sh
+or with your app running locally but connected to deployed services.
+In either case, make sure to debug in a non-production environment.
 
 {{% troubleshoot %}}
 
-## Remote debugging
+## 1. Create a new environment
 
-### Create a new environment
+Start by creating a new environment completely isolated from production but with the same data for debugging:
 
-```sh
-$ platform branch debug-branch
+```bash
+platform branch debug-branch
 ```
 
-### SSH into your app container
+## 2. Get access
 
-```sh
-$ platform ssh
+{{< codetabs >}}
+
+---
+title=Remote
+file=none
+highlight=false
+---
+
+Access your app container via [SSH](../../development/ssh/_index.md):
+
+```bash
+platform ssh
 ```
 
-In a multi-application project, you will be prompted to chose which application to SSH into. You can also specify the application you want using the `--app` option:
+<--->
+
+---
+title=Local
+file=none
+highlight=false
+---
 
-```sh
-$ platform ssh --app=my-app-name
+To access deployed apps and services, open tunnels to everything your app has relationships with:
+
+```bash
+platform tunnel:open
+```
+
+In the same terminal, set the relevant environment variables:
+
+```bash
+export PLATFORM_RELATIONSHIPS="$(platform tunnel:info --encode)"
+export PORT=8888
 ```
+<!--vale off -->
+{{< /codetabs >}}
+<!--vale on -->
 
-### Restart node in inspect mode 
 
-Once connected to your app container, run:
+## 3. Run your app in inspect mode
 
-```sh
-$ sv stop <app_name>
-$ node --inspect <js_file>
+{{< codetabs >}}
+
+---
+title=Remote
+file=none
+highlight=false
+---
+
+Stop the current process and restart it in inspect mode:
+
+```bash
+sv stop app
+node --inspect <START_FILE>
 ```
 
-`app_name` being the name of your application as defined in your `.platform.app.yaml` and `js_file` being your application entrypoint.
+Replace `<START_FILE>` with the file defined for [your app's `start` command](./_index.md#4-start-your-app).
 
-You should see something like:
+You get output something like this:
```

Comment:
:pick:
```suggestion
You get an output like:
```

Reply:
I wouldn't tread `output` as a countable noun in this situation.